Backend/Microservices/ChatService/src/Chat.Data/Services/CacheService.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Chat.Application.DTOs;
+using Chat.Application.Interfaces;
+using Chat.Domain.Entities;
+using Microsoft.Extensions.Caching.Distributed;
+using System.Text.Json;
+
+namespace Chat.Data.Services;
+
+public class CacheService : ICacheService
+{
+    private readonly IUserRepository _userRepository;
+    private readonly IMessageRepository _messageRepository;
+    private readonly IDistributedCache _cache;
+
+    public CacheService(IUserRepository userRepository, IMessageRepository messageRepository, IDistributedCache cache)
+    {
+        _userRepository = userRepository;
+        _messageRepository = messageRepository;
+        _cache = cache;
+    }
+
+    public async Task CachingConnection(string connectionId, UserConnection connection)
+    {
+        var stringConnection = JsonSerializer.Serialize(connection);
+        await _cache.SetStringAsync(connectionId, stringConnection);
+    }
+
+    public async Task<UserConnection?> GetConnectionFromCache(string connectionId)
```

Comment:
I think FromCache in the function name is redundant, the class/object name already points to the cache